ReactiveCocoa/Swift/UnidirectionalBinding.swift

```diff
@@ -0,0 +1,313 @@
+import enum Result.NoError
+
+precedencegroup BindingPrecedence {
+	associativity: right
+
+	// Binds tighter than assignment but looser than everything else
+	higherThan: AssignmentPrecedence
+}
+
+infix operator <~ : BindingPrecedence
+
+/// Describes a target to which can be bound.
+public protocol BindingTarget: class {
+	associatedtype ValueType
```

Comment:
Anyway, it was made a workaround as sometimes somehow the type parameters cannot be matched for associated type requirements. It seems it was solved by reordering the protocol inheritance clause...